apps/ledger-live-desktop/src/renderer/hooks/usePortfolioCards.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import ActionCard from "~/renderer/components/ContentCards/ActionCard";
+import { portfolioContentCardSelector } from "~/renderer/reducers/dynamicContent";
+import * as braze from "@braze/web-sdk";
+import { setPortfolioCards } from "~/renderer/actions/dynamicContent";
+
+const usePortfolioCards = () => {
+  const dispatch = useDispatch();
+  const [cachedContentCards, setCachedContentCards] = useState<braze.Card[]>([]);
+  const portfolioCards = useSelector(portfolioContentCardSelector);
+
+  useEffect(() => {
+    const cards = braze.getCachedContentCards().cards;
+    setCachedContentCards(cards);
+  }, []);
+
+  const onImpression = (cardId: string) => {
+    const currentCard = cachedContentCards.find(card => card.id === cardId);
+    currentCard && braze.logContentCardImpressions([currentCard]);
+  };
+
+  const onDismiss = (cardId: string) => {
+    const currentCard = cachedContentCards.find(card => card.id === cardId);
+
+    if (currentCard) {
+      braze.logCardDismissal(currentCard);
+      setCachedContentCards(cachedContentCards.filter(n => n.id !== currentCard.id));
+      dispatch(setPortfolioCards(portfolioCards.filter(n => n.id !== currentCard.id)));
+    }
+  };
+
+  const onClick = (cardId: string) => {
+    const currentCard = cachedContentCards.find(card => card.id === cardId);
```

Comment:
As use it many times you can create someting like on top and use it everywhere ?
```suggestion
    const getCurrentCard = (cardId) => cachedContentCards.find(card => card.id === cardId);
```